src/huggingface_hub/hf_api.py

```diff
@@ -5760,8 +5760,18 @@ def create_branch(
         try:
             hf_raise_for_status(response)
         except HfHubHTTPError as e:
-            if not (e.response.status_code == 409 and exist_ok):
-                raise
+            if e.response.status_code == 409 and exist_ok:
```

Comment:
```suggestion
            if exist_ok and e.response.status_code == 409:
```
reverse order for symmetry with the line below?